adabot/lib/circuitpython_library_validators.py

```diff
@@ -257,6 +258,29 @@ def validate_repo_state(self, repo):
                 errors.append(ERROR_ONLY_ALLOW_MERGES)
         return errors
 
+    def validate_actions_state(self, repo):
+        """Validate if the most recent GitHub Actions run on the default branch
+        has passed.
+        Just returns a message stating that the most recent run failed.
+        """
+        if not (repo["owner"]["login"] == "adafruit" and
+                repo["name"].startswith("Adafruit_CircuitPython")):
+            return []
+        
+        main_branch = repo["default_branch"]
+        response = github.get("/repos/" + repo["full_name"] + "/actions/runs")
+        if not response.ok:
+            return [error_unable_pull_repo_details]
+        
+        workflow_runs = response.json()["workflow_runs"]
+        for workflow in workflow_runs:
+            if workflow["head_branch"] == main_branch:
+                if workflow["conclusion"] == "failure":
+                    return [ERROR_GITHUB_FAILING_ACTIONS]
+                break
```

Comment:
This logic relies on the `/actions/runs` endpoint returning the runs in reverse chronological order. The API docs don't state the ordering, so its kind of unknown.

Checking the latest status _could_ achieve this too. The [docs](https://docs.github.com/en/rest/reference/repos#list-commit-statuses-for-a-reference) state that the results are returned in reverse chronological order, making this a known truth. I tried pulling some requests, but didn't get anything returned. _(I tried via browser, so that could be the issue due to lack of a specific request header. Or, the endpoint may be disabled. 🤷)_

If the request endpoint isn't changed, and the suggestion above to limit the branch is done, you could simply check the first item in the list for the "latest" run's result.
```python
workflow_runs = response.json()["workflow_runs"]
if workflow_runs[0]["conclusion"] == failure:
   return [ERROR_GITHUB_FAILING_ACTIONS]
```